src/tech/hlsjs.js

```diff
@@ -1,31 +1,85 @@
 import videojs from 'video.js/core';
 import Hls from "hls.js";
 
+/** Follow up to ask for error type exports on VJS */
+/**
+ * Errors indexed by the W3C standard. The order **CANNOT CHANGE**! See the
+ * specification listed under {@link MediaError} for more information.
+ *
+ * @enum {array}
+ * @readonly
+ * @property {string} 0 - MEDIA_ERR_CUSTOM
+ * @property {string} 1 - MEDIA_ERR_ABORTED
+ * @property {string} 2 - MEDIA_ERR_NETWORK
+ * @property {string} 3 - MEDIA_ERR_DECODE
+ * @property {string} 4 - MEDIA_ERR_SRC_NOT_SUPPORTED
+ * @property {string} 5 - MEDIA_ERR_ENCRYPTED
+ */
+const VJSErrorTypes = {
+  MEDIA_ERR_CUSTOM: 0, 
+  MEDIA_ERR_ABORTED: 1, 
+  MEDIA_ERR_NETWORK: 2, 
+  MEDIA_ERR_DECODE: 3, 
+  MEDIA_ERR_SRC_NOT_SUPPORTED: 4, 
+  MEDIA_ERR_ENCRYPTED: 5
+};
+
+const ErrorTypeMap = {
+  [Hls.ErrorTypes.NETWORK_ERROR]: VJSErrorTypes.MEDIA_ERR_NETWORK,
+  [Hls.ErrorTypes.MEDIA_ERROR]: VJSErrorTypes.MEDIA_ERR_DECODE,
+  [Hls.ErrorTypes.MUX_ERROR]: VJSErrorTypes.MEDIA_ERR_DECODE,
+  [Hls.ErrorTypes.KEY_SYSTEM_ERROR]: VJSErrorTypes.MEDIA_ERR_ENCRYPTED,
+  DEFAULT: VJSErrorTypes.MEDIA_ERR_CUSTOM,
+};
+
+// NOTE: Since we actually want our tech to be something in between the Html5 tech and the vhs tech, we need to grab this to support the
+// built in error management/pass-through from "`Tech`" (instead of assuming it will always/only be an HTMLMediaElement error, which is what the Html5 tech assumes).
+const baseTechError = videojs.getTech('Tech').prototype.error;
+
 class HlsJs {
   static hlsTypeRE = /^application\/(x-mpegURL|vnd\.apple\.mpegURL)$/i;
   static hlsExtRE = /\.m3u8/i;
 
   constructor(source, tech, options = {}) {
     this.source = source;
     this.tech = tech;
+    // Use the base `Tech` error implementation instead of the Html5 one.
+    this.tech.error = baseTechError;
```

Comment:
Interestingly, the way VHS does this by grabbing a reference to the player and then doing `player.error({code: 4})` (or whatever the appropriate code is). Which isn't necessarily the best method.
This does bring up that in Video.js we need either
1. bring back a `player` reference inside of tech, or
2. have proper way of bubbling errors from source handlers to the player
Since if you are creating a full on Tech, rather than a source handler, then you can rely on having the error be available on `tech.error()` and then you trigger an `error` event on the tech.

Reply:
> 1. **_bring back_** a `player` reference inside of tech, or

That must've been buried away in my memory, since I started hunting for a player reference earlier on in the implementation.

Reply:
> An alternative to this method, particuarly since you already retrigger all the hlsjs events on the player is to create a player plugin that calls `player.error()`

Ah I wasn't confident that there weren't Other Things™️ that happened internally as a result of the errors being triggered from the `tech` instance (as opposed to hop-scotching directly to the `player` instance's public API).